### PR TITLE
fix: download csv column order

### PR DESCRIPTION
**Pull request by rephus**
Closes: https://github.com/lightdash/lightdash/issues/7753

### Description:

Screenshot from 2023-11-07 15-34-47
Screenshot from 2023-11-07 15-24-21

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx [hunk 1]
```diff
@@ -108,52 +109,18 @@ const VisualizationCard: FC<{
...
-            let filteredColumnOrder: string[] | undefined;
-
-            if (
-                unsavedChartVersion.chartConfig.type === 'table' &&
-                unsavedChartVersion.chartConfig.config?.columns
-            ) {
-                const columnsConfig =
-                    unsavedChartVersion.chartConfig.config?.columns;
-                filteredColumns =
-                    unsavedChartVersion?.metricQuery.dimensions.filter(
```
Comment (rephus) [inline packages/frontend/src/components/Explorer/VisualizationCard/VisualizationCard.tsx:121]: the old logic was wrong, it was not considering `visible` metrics

File: packages/frontend/src/utils/csvUtils.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        const filterFieldNotVisible = (fieldId: string) =>
+            !filteredColumnIds?.includes(fieldId);
+
+        return {
+            metricQuery: {
+                ...metricQuery,
+                metrics: metricQuery.metrics.filter(filterFieldNotVisible),
+                dimensions: metricQuery.dimensions.filter(
+                    filterFieldNotVisible,
+                ),
```
Comment (ZeRego) [inline packages/frontend/src/utils/csvUtils.ts:29]: I don't think we should manipulate the metricQuery values.
What happens if a table calculation depends on the dimension/metric?

Comment (rephus) [reply packages/frontend/src/utils/csvUtils.ts:29]: yeah, great point, instead we should pass the table config and hide the columns on the csv generation, right ? 


I can simply remove the filtering to fix https://github.com/lightdash/lightdash/issues/7753 but not https://github.com/lightdash/lightdash/issues/7702

**Comment by rephus** [comment]
Reverted "filter" code , removed old filtering to fix sorting

image

Fixing visibility will be done in a separate PR

**Review by ZeRego — APPROVED** [review]

**Pull request merged**